### PR TITLE
remove autoloading of extras.wad

### LINKED ISSUE #2073: Change extras.wad to be loaded before the IWAD, or don't load it for non-Id IWADs.
Loading extras.wad as it's loaded currently may work perfectly fine for Doom 1, Doom 2 and Final Doom but for non-Id IWADs such as Chex Quest and Freedoom it seems to cause some issues. Two examples of issues this causes can be demonstrated in this screenshot of Chex Quest 3: Vanilla Edition:

Woof0000

Chex 3 Vanilla provides its own graphics for the weapon wheel icons (as of 1.0) and having both keys, but then extras.wad is loaded after it and replaces the proper Chex Quest weapon/key icons with the Doom ones, which doesn't make any sense here.

To fix this issue, I suggest doing one of two things: Either load extras.wad right before the IWAD instead of after, in order to allow the IWAD to override its resources (which shouldn't affect the Id IWADs since they don't include that stuff in the actual IWAD anyways) or add a check to only load extras.wad if the IWAD being used is an Id IWAD (so only if it's Doom 1, Doom 2 or Final Doom). It's also worth noting that the former option is in fact how KEX Doom does it, too.

### PR DESCRIPTION
**Pull request by rfomin**
Fix #2073

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
Will checks like these still work?

https://github.com/search?q=repo%3Afabiangreffrath%2Fwoof%20wadfiles%5B0%5D&type=code

**Comment by fabiangreffrath** [comment]
Why not load extras.wad only for Doom games?

https://github.com/fabiangreffrath/woof/blob/c7feec0a600bfbfbd62b9ed88aa47f19a232ec69/src/d_main.c#L1485

**Comment by rfomin** [comment]
> Will checks like these still work?

I forgot about these.

> Why not load extras.wad only for Doom games?

For example, chex3v.wad is detected as `gamession == doom`. We need a better mechanism to detect vanilla IWADs (calculate hash of some lumos?). Perhaps it's better to remove extras.wad autoloading for now.

**Review by fabiangreffrath — APPROVED** [review]

File: src/d_main.c [hunk 1]
```diff
@@ -1834,21 +1834,6 @@ void D_DoomMain(void)
   // [FG] emulate a specific version of Doom
   InitGameVersion();
 
-  //!
-  // @category mod
-  //
-  // Disable auto loading of extras.wad.
-  //
-
-  if (!M_ParmExists("-noextras"))
```
Comment (fabiangreffrath) [inline src/d_main.c:1843]: The parameter ist still in `params.h`.

Comment (rfomin) [reply src/d_main.c:1843]: Fixed in f88fe30

**Pull request merged**